src/app/api/volunteer/auth/sign-up/route.ts

```diff
@@ -0,0 +1,63 @@
+import { Role } from "@/common_utils/types";
+import { getUserByEmail, volunteerSignUp } from "@server/mongodb/actions/User";
+import APIWrapper from "@server/utils/APIWrapper";
+
+type SignupData = {
+  email: string;
+  name: string;
+  phoneNumber: string;
+  country: string;
+  state: string;
+  city: string;
+  chapter: string;
+};
+
+export const POST = APIWrapper({
+  config: {
+    requireToken: true,
+    roles: [Role.NONPROFIT_ADMIN, Role.NONPROFIT_VOLUNTEER],
+  },
+  handler: async (req: Request) => {
+    const signupData = (await req.json()) as SignupData;
+
+    if (!signupData) {
+      throw new Error("missing request data");
+    }
+
+    if (
+      !signupData.chapter ||
+      !signupData.city ||
+      !signupData.country ||
+      !signupData.email ||
+      !signupData.name ||
+      !signupData.phoneNumber ||
+      !signupData.state
+    ) {
+      throw new Error("missing parameter(s)");
```

Comment:
Can you change "missing" -> "Missing". It's good to use capitalization in these messages.